### PR TITLE
Amazon test cleanup

### PR DESCRIPTION
**Pull request by pankajastro**
closes: https://github.com/astronomer/astronomer-providers/issues/690
related-to:  https://github.com/astronomer/astronomer-providers/issues/568

- convert the function-based test to a class-based 
- remove context fixture definition from operator/sensor/extractor and reuse from conftest
- consolidate EMR sensor tests into a single test file 
- remove function `create_context` from a couple of test files and reuse it from the util

### CODE REVIEW

File: tests/amazon/aws/extractors/test_redshift.py [hunk 1]
```diff
@@ -1,94 +1,40 @@
...
-
-@mock.patch("airflow.models.taskinstance.TaskInstance")
-def test__get_xcom_redshift_job_id(mock_task_instance):
-    mock_task_instance.xcom_pull.return_value = ["123"]
-    extractor = RedshiftAsyncExtractor(TASK)
-    redshift_job_id = extractor._get_xcom_redshift_job_id(mock_task_instance)
-    assert "123" == redshift_job_id
+class TestRedshiftAsyncExtractor:
+    def test_get_operator_classnames(self):
+        extractor = RedshiftAsyncExtractor(TASK)
```
Comment (bharanidharan14) [inline tests/amazon/aws/extractors/test_redshift.py:24]: TASK constant can be inside the class

File: tests/amazon/aws/hooks/test_batch_client.py [hunk 2]
```diff
@@ -10,129 +10,132 @@
...
+    @pytest.mark.asyncio
+    @mock.patch("astronomer.providers.amazon.aws.hooks.batch_client.BatchClientHookAsync.get_client_async")
+    @mock.patch("astronomer.providers.amazon.aws.hooks.batch_client.BatchClientHookAsync.poll_job_status")
+    async def test_monitor_job_with_success(self, mock_poll_job_status, mock_client):
+        """Tests that the  monitor_job method returns expected event once successful"""
+        mock_poll_job_status.return_value = True
+        mock_client.return_value.__aenter__.return_value.describe_jobs.return_value = (
+            BATCH_API_SUCCESS_RESPONSE
+        )
+        hook = BatchClientHookAsync(job_id=JOB_ID, waiters=None)
```
Comment (bharanidharan14) [inline tests/amazon/aws/hooks/test_batch_client.py:23]: JOB_ID Global constant can be moved inside the class as class level variable

**Review by bharanidharan14 — COMMENTED** [review]
Overall looks good, all global constant can be moved inside the class as class level variable

File: tests/amazon/aws/operators/test_batch.py [hunk 3]
```diff
@@ -20,100 +18,72 @@
...
+        with pytest.raises(TaskDeferred) as exc:
+            task.execute(context)
+        assert isinstance(exc.value.trigger, BatchOperatorTrigger), "Trigger is not a BatchOperatorTrigger"
+
+    def test_batch_op_async_execute_failure(self, context):
+        """Tests that an AirflowException is raised in case of error event"""
+
+        task = BatchOperatorAsync(
+            task_id="task",
+            job_name=JOB_NAME,
```
Comment (bharanidharan14) [inline tests/amazon/aws/operators/test_batch.py:49]: Same here

File: tests/amazon/aws/operators/test_batch.py [hunk 4]
```diff
@@ -20,100 +18,72 @@
...
+            job_queue="queue",
+            job_definition="hello-world",
+            max_retries=MAX_RETRIES,
```
Comment (bharanidharan14) [inline tests/amazon/aws/operators/test_batch.py:52]: Applies to all the global constant

File: tests/amazon/aws/sensors/test_emr_sensors.py [hunk 5]
```diff
@@ -1,131 +1,164 @@
...
     EmrJobFlowSensorAsync,
+    EmrStepSensorAsync,
 )
 from astronomer.providers.amazon.aws.triggers.emr import (
     EmrContainerSensorTrigger,
     EmrJobFlowSensorTrigger,
+    EmrStepSensorTrigger,
 )
 
 TASK_ID = "test_emr_container_sensor"
```
Comment (bharanidharan14) [inline tests/amazon/aws/sensors/test_emr_sensors.py:17]: Same here

**Comment by rajaths010494** [comment]
tests/amazon/aws/sensors/test_emr.py why is this file deleted? 
and where is the corresponding test cases for it?

File: tests/amazon/aws/extractors/test_redshift.py (hunk 1 above)
Comment (rajaths010494) [reply tests/amazon/aws/extractors/test_redshift.py:24]: if we move this to inside class make this consistent across all remaining test cases. (eg in apache, google, amazon and all others remaining)

**Comment by pankajastro** [comment]
> tests/amazon/aws/sensors/test_emr.py why is this file deleted? and where is the corresponding test cases for it?

Moved it `test_emr_sensors.py`

File: tests/amazon/aws/triggers/test_s3_triggers.py [hunk 6]
```diff
@@ -11,152 +11,149 @@
...
-        task = asyncio.create_task(trigger.run().__anext__())
-        await asyncio.sleep(0.5)
+class TestS3KeyTrigger:
+    def test_s3_key_trigger_serialization(self):
+        """
+        Asserts that the TaskStateTrigger correctly serializes its arguments
+        and classpath.
+        """
+        trigger = S3KeyTrigger(
+            bucket_key="s3://test_bucket/file", bucket_name="test_bucket", wildcard_match=True
```
Comment (rajaths010494) [inline tests/amazon/aws/triggers/test_s3_triggers.py:21]: `bucket_key` and `bucket_name`  value  can be made as Constant since all tests cases use it.

**Comment by bharanidharan14** [comment]
@pankajastro can we take up this?

**Review by rajaths010494 — APPROVED** [review]

**Comment by rajaths010494** [comment]
Rebase the branch.

**Comment by rajaths010494** [comment]
Thanks @pankajastro  for changing the amazon related test cases to class based it was a huge one.

**Pull request merged**